binaries/cuprated/src/rpc/handler.rs

```diff
@@ -54,6 +56,61 @@ pub enum BlockchainManagerRequest {
 
     /// The height of the next block in the chain.
     TargetHeight,
+
+    /// Calculate proof-of-work for this block.
+    CalculatePow {
+        /// The hardfork of the protocol at this block height.
+        hardfork: HardFork,
+        /// The height of the block.
+        height: usize,
+        /// The block data.
+        block: Block,
+        /// The seed hash for the proof-of-work.
+        seed_hash: [u8; 32],
+    },
+
+    /// Add auxirilly proof-of-work to a block.
+    ///
+    /// From the RPC `add_aux_pow` usecase's documentation:
+    /// ````
+    /// This enables merge mining with Monero without requiring
+    /// software that manually alters the extra field in the coinbase
+    /// tx to include the merkle root of the aux blocks.
+    /// ````
+    AddAuxPow {
+        /// The block template to add to.
+        block_template: Block,
+        /// The auxirilly proof-of-work to add.
+        aux_pow: Vec<AuxPow>,
+    },
+
+    /// Generate new blocks.
+    ///
+    /// This request is only for regtest, see RPC's `generateblocks`.
+    GenerateBlocks {
+        /// Number of the blocks to be generated.
+        amount_of_blocks: u64,
+        /// The previous block's hash.
+        prev_block: [u8; 32],
+        /// The starting value for the nonce.
+        starting_nonce: u32,
+        /// The address that will receive the coinbase reward.
+        wallet_address: String,
+    },
+
+    /// Get a visual [`String`] overview of blockchain progress.
+    ///
+    /// This is a highly implementation specific format used by
+    /// `monerod` in the `sync_info` RPC call's `overview` field;
+    /// it is essentially an ASCII visual of blocks.
+    ///
+    /// See also:
+    /// - <https://www.getmonero.org/resources/developer-guides/daemon-rpc.html#sync_info>
+    /// - <https://github.com/monero-project/monero/blob/master/src/cryptonote_protocol/block_queue.cpp#L178>
+    Overview {
```

Comment:
### TODO
Is it worth re-implementing and supporting these highly specific one-off `monerod` string formats? There's more than this one too.

Reply:
@Boog900 I agree let's do put a surprise there